packages/experiments/src/components/MicroFeedback/MicroFeedback.view.tsx

```diff
@@ -0,0 +1,128 @@
+
+export const MicroFeedbackView: IMicroFeedbackComponent['view'] = props => {
+  const {
+    children,
+    sendFollowUpIndex,
+    likeIconTitle,
+    dislikeIconTitle,
+    likeQuestion,
+    dislikeQuestion,
+    vote,
+    isFollowUpVisible,
+    isThanksVisible,
+    likeRef,
+    dislikeRef,
+    onCalloutDismiss,
+    onThanksDismiss,
+    onThanksShow,
+    onLikeVote,
+    onDislikeVote,
+    thanksText,
+    inline
+  } = props;
+
+  const likeIcon = vote === 'like' ? 'LikeSolid' : 'Like';
+  const dislikeIcon = vote === 'dislike' ? 'DislikeSolid' : 'Dislike';
+  const hideDislikeCallout = vote !== 'dislike' || !isFollowUpVisible;
+  const hideLikeCallout = vote !== 'like' || !isFollowUpVisible;
+
+  const Slots = getSlots<IMicroFeedbackProps, IMicroFeedbackSlots>(props, {
+    root: Stack,
+    iconContainer: Stack,
+    followUpContainer: inline ? Stack : Callout,
+    followUpQuestion: Text,
+    followUpOptionList: List,
+    followUpOption: Button,
+    followUpOptionText: Text,
+    thanksContainer: Callout
+  });
+
+  const followUpOptionTokens: IButtonTokens = { contentPadding: '6px 0px' };
+
+  const onRenderFollowup = (followUp: IMicroFeedbackQuestion, targetRef: HTMLDivElement | null): JSX.Element => {
+    const onRenderCalloutItem = (item: string, index: number | undefined): JSX.Element => {
+      const listOption = (): void => {
+        onCalloutDismiss();
+        if (sendFollowUpIndex && index !== undefined && followUp) {
+          sendFollowUpIndex(followUp!.id, index);
+          onThanksShow();
+        }
+      };
+
+      return (
+        <Slots.followUpOption data-is-focusable={true} onClick={listOption} tokens={followUpOptionTokens}>
+          <Slots.followUpOptionText>{`${item}`}</Slots.followUpOptionText>
+        </Slots.followUpOption>
+      );
+    };
+
+    return (
+      <Slots.followUpContainer gapSpace={0} onDismiss={onCalloutDismiss} role="alertdialog" setInitialFocus={true} target={targetRef}>
+        <FocusZone direction={FocusZoneDirection.vertical}>
+          <Slots.followUpQuestion block variant="small">
+            {followUp.question}
+          </Slots.followUpQuestion>
+          <Slots.followUpOptionList items={followUp.options} onRenderCell={onRenderCalloutItem} />
+        </FocusZone>
+      </Slots.followUpContainer>
+    );
+  };
+
+  const showThanks = (followUp: IMicroFeedbackQuestion | undefined, voteType: string): void => {
+    // If vote while thanks is showing, dismiss
+    if (isThanksVisible) {
+      onThanksDismiss();
+    }
+
+    // Show thanks if there is no follow up question and not  unselecting a vote
+    if (!followUp && vote !== voteType) {
+      onThanksShow();
+    }
+  };
+
+  const likeVoteClick = (): void => {
+    showThanks(likeQuestion, 'like');
+    onLikeVote();
+  };
+
+  const dislikeVoteClick = (): void => {
+    showThanks(dislikeQuestion, 'dislike');
+    onDislikeVote();
+  };
```

Comment:
Lots of functions being created on every render pass here, but we can mitigate with `useCallback` when merged into `fabric-7`.

Reply:
When would this need to be converted?

Reply:
When it gets merged into the `fabric-7` branch. It can't be converted on master since it requires a newer version of React.